utils/crypto_utils.h

```diff
@@ -19,12 +19,12 @@ extern "C" {
 #define MAXLINE 1024
 #define IMSI_LEN 15
 
-int get_device_id(const char *device_id);
-int get_aes_key(const uint8_t *key);
-int encrypt(unsigned char *plaintext, int plaintext_len, unsigned char *ciphertext, int ciphertext_len, uint8_t iv[16],
-            uint8_t key[AES_BLOCK_SIZE * 2], uint8_t device_id[IMSI_LEN + 1]);
-int decrypt(unsigned char *ciphertext, int ciphertext_len, unsigned char *plaintext, int plaintext_len, uint8_t iv[16],
-            uint8_t key[AES_BLOCK_SIZE * 2]);
+int get_device_id(char *device_id);
+int get_aes_key(uint8_t *key);
+int ta_encrypt(unsigned char *plaintext, int plaintext_len, unsigned char *ciphertext, int ciphertext_len,
+               uint8_t iv[16], uint8_t key[AES_BLOCK_SIZE * 2], uint8_t device_id[IMSI_LEN + 1]);
+int ta_decrypt(unsigned char *ciphertext, int ciphertext_len, unsigned char *plaintext, int plaintext_len,
+               uint8_t iv[16], uint8_t key[AES_BLOCK_SIZE * 2]);
```

Comment:
May you assign the size of IV with macro `AES_IV_SIZE` and define one more macro for `AES_KEY_SIZE`?
Btw are we using aes cbc here? If we are using cbc, then I think it is better to name the macro of iv size as `CBC_IV_SIZE`